lib/lte_link_control/lte_lc.c

```diff
@@ -1802,6 +1802,11 @@ int lte_lc_factory_reset(enum lte_lc_factory_reset_type type)
 	return nrf_modem_at_printf("AT%%XFACTORYRESET=%d", type) ? -EFAULT : 0;
 }
 
+#define LTE_LC_EVENT_STACK_SIZE 1024 /* TODO: Make size Kconfig option */
+K_THREAD_STACK_DEFINE(lte_lc_event_stack, LTE_LC_EVENT_STACK_SIZE);
+
+struct k_work_q lte_lc_event_work_q;
```

Comment:
I'm not sure if we need a new workqueue but using the existing one where AT commands are sent may be troublesome too.

Reply:
I'm afraid we need to use a dedicated workqueue to avoid getting stuck when AT%XMONITOR or AT%XPTW is sent.